glean_parser/templates/kotlin.jinja2

```diff
@@ -64,4 +67,18 @@ internal object {{ category_name|Camelize }} {
     {{ obj_declaration(obj, lazy=obj.type != 'ping') }}
     {% endif %}
 {%- endfor %}
+{% if has_gecko_datapoints %}
+    // Support exfiltration of Gecko metrics from products using both the
+    // Glean SDK and GeckoView. See bug 1566356 for more context.
+    operator fun get(geckoMetricName: String): HistogramBase? {
+        return when(geckoMetricName) {
+        {% for obj in objs.values() %}
+            {% if obj.gecko_datapoint %}
+            "{{ obj.gecko_datapoint }}" -> {{ obj.name|camelize }}
+            {% endif %}
+        {%- endfor %}
+            else -> null
+        }
+    }
+{% endif %}
```

Comment:
This is fine, but wanted to call out that since each category of metrics is written to its own `.kt` file, this function will only include the mapping for a particular category.  Maybe that's sufficient for the use case, of course.

If we needed a global mapping of everything, we'd probably need to use a technique similar to how pings are registered -- i.e. a singleton data structure that is updated at startup time.

Reply:
As we discussed over Slack, let's take this as follow-up work since it's not blocking the current batch of metrics. I filed [this bug](https://bugzilla.mozilla.org/show_bug.cgi?id=1567894).